lib/backbeat/Metrics.js

```diff
@@ -208,33 +209,43 @@ class Metrics {
             const now = new Date();
             const timeSinceStart = Math.floor(
                 (now - this._internalStart) / 1000);
+            // Seconds up to a max of EXPIRY seconds
+            const timeDisplay = timeSinceStart < EXPIRY ?
+                timeSinceStart : EXPIRY;
+            const numOfIntervals = Math.ceil(timeDisplay / INTERVAL);
 
             const [opsThroughput, bytesThroughput] = res.map(r => {
-                let total = r.requests.slice(0, 3).reduce(
+                let total = r.requests.slice(0, numOfIntervals).reduce(
                     (acc, i) => acc + i, 0);
 
-                // last interval timestamp
-                const lastInterval =
-                    this._statsClient._normalizeTimestamp(new Date(now));
-                // in seconds
-                const diff = Math.floor((now - lastInterval) / 1000);
+                // if timeDisplay !== EXPIRY, use internal timer and do not
+                // include the extra 4th interval
+                if (timeDisplay === EXPIRY) {
+                    // all intervals apply, including 4th interval
+                    const lastInterval =
+                        this._statsClient._normalizeTimestamp(new Date(now));
+                    // in seconds
+                    const diff = Math.floor((now - lastInterval) / 1000);
 
-                if (timeSinceStart >= EXPIRY) {
-                    // Get average for last interval depending on time surpassed
-                    // so far for newest interval
-                    total += ((INTERVAL - diff) / INTERVAL) * r.requests[3];
+                    if (timeSinceStart >= EXPIRY) {
```

Comment:
Maybe missing something, but how is this test different from testing `timeDisplay === EXPIRY`?